src/backend/commands/copy.c

```diff
@@ -7955,9 +7955,15 @@ GetTargetSeg(GpDistributionData *distData, Datum *baseValues, bool *baseNulls)
 	return target_seg;
 }
 
+static void close_program_pipes_on_reset(void *arg) {
+	CopyState cstate = arg;
+	close_program_pipes(cstate, !IsAbortInProgress());
```

Comment:
`close_program_pipes` may throw `elog(ERROR`. Is it valid to throw errors during memory context cleanup?

Reply:
I changed it so that the callback would trigger cleanup only on errors.